src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -1554,6 +1554,57 @@ public void TestWithSyntaxTree()
             var root = newTree.GetRoot();
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
+        [WorkItem(3705, "https://github.com/dotnet/roslyn/issues/3705")]
+        public void TestAnalyzerConfigOptions()
+        {
+            // get one to get to syntax tree factory
+            var dummyProject = CreateNotKeptAliveSolution().AddProject("dummy", "dummy", LanguageNames.CSharp);
+
+            var factory = dummyProject.LanguageServices.SyntaxTreeFactory;
+
+            // create the origin tree
+            var testDir = Temp.CreateDirectory();
+            var sourceFile = testDir.CreateFile("test.cs").WriteAllText(@"
+class C
+{
+    void M(C? c)
+    {
+        _ = c.ToString();   // warning CS8602: Dereference of a possibly null reference.
+    }
+}");
+            using var fileStream = File.OpenRead(sourceFile.Path);
+            var text = SourceText.From(fileStream);
+            var tree = factory.ParseSyntaxTree(sourceFile.Path, dummyProject.ParseOptions, text, analyzerConfigOptionsResult: null, CancellationToken.None);
```

Comment:
Why is this test in SolutionTests? It's not really testing anything about the Workspace at all since it's directly using primitives. Why not add a source file to the project snapshot you create, let the workspace make the compilation and use that? It also confirms that adding an .editorconfig to the snapshot will force us to make a new syntax tree.

Reply:
(see all the tests later in the file)

Reply:
Ah, let me rewrite this test.